### PR TITLE
GetInteriorVehicleDataCapabilities API of Reverse SDL

### PR DESCRIPTION
**Pull request by dev-gh**
Changes contain 
- working RSDL build
- basic implementation of GetInteriorVehicleDataCapabilities RPC
- some RPCs have been disabled according to customer proposal (https://github.com/yang1070/sdl_evolution/blob/170b85f894fc64f763b6c600708ff5683004ac89/proposals/NNNN-remote-control-baseline.md)
- code style changes with clang
- bugfixes for REMOTE_CONTROL enum conversion, RC RPCs permissions validation, GetInteriorVehicleDataCapabilities response validation

### CODE REVIEW

File: src/components/can_cooperation/include/can_cooperation/commands/get_interior_vehicle_data_capabilities_request.h [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+ protected:
+  virtual application_manager::TypeAccess CheckAccess(
+      const Json::Value& message);
+  virtual std::string ModuleType(const Json::Value& message);
+
+ private:
+  void UpdateModules(Json::Value* params);
+  application_manager::TypeAccess CheckModuleTypes(const Json::Value& message);
+  bool ReadCapabilitiesFromFile();
+  void CreateCapabilities(const Json::Value& zone_capabilities,
```
Comment (AByzhynar) [inline src/components/can_cooperation/include/can_cooperation/commands/get_interior_vehicle_data_capabilities_request.h:90]: @dev-gh `zone_capabilities`? Are you sure that we have to handle this parameter?

**Review by AByzhynar — CHANGES_REQUESTED** [review]

File: src/components/can_cooperation/src/commands/get_interior_vehicle_data_capabilities_request.cc [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+      trying to read from file");
+  Json::Value request;
+  Json::Reader reader;
+  if (!reader.parse(message_->json_message(), request)) {
+    LOG4CXX_ERROR(logger_, "Failed to read capabilities from file also");
+    return false;
+  }
+
+  VehicleCapabilities file_caps;
+  Json::Value zone_capabilities;
```
Comment (AByzhynar) [inline src/components/can_cooperation/src/commands/get_interior_vehicle_data_capabilities_request.cc:161]: @dev-gh The same question as above: `Zone_capabilities`? Do we need it?

**Review by AByzhynar — CHANGES_REQUESTED** [review]

**Review by AKalinich-Luxoft — APPROVED** [review]

File: src/components/can_cooperation/src/validators/get_interior_vehicle_data_capabilities_response_validator.cc [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+  // Stub logic, add validation of climate and radio structs
+  outgoing_json = json;
+
+  return result;
+}
+
+}  // namespace valdiators
+
+}  // namespace can_cooperation
```
Comment (AByzhynar) [inline src/components/can_cooperation/src/validators/get_interior_vehicle_data_capabilities_response_validator.cc]: @dev-gh Please add one empty line at the end of all files

**Review by AByzhynar — CHANGES_REQUESTED** [review]

**Comment by dev-gh** [comment]
@LuxoftAKutsan @LitvinenkoIra 
Please review

File: src/components/application_manager/include/application_manager/message.h [hunk 4]
```diff
@@ -92,7 +92,8 @@ class Message {
   size_t payload_size() const;
   const smart_objects::SmartObject& smart_object() const;
 
-  //! --------------------------------------------------------------------------.
+  //!
+  //--------------------------------------------------------------------------.
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/include/application_manager/message.h]: @dev-gh that does this changes mean?

File: src/components/application_manager/test/include/application_manager/mock_application.h [hunk 5]
```diff
@@ -285,8 +285,9 @@ class MockApplication : public ::application_manager::Application {
   MOCK_CONST_METHOD0(PackageName, std::string());
 
 #ifdef SDL_REMOTE_CONTROL
-  MOCK_METHOD1(set_system_context,
-               void(const application_manager::mobile_api::SystemContext::eType&));
+  MOCK_METHOD1(
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/test/include/application_manager/mock_application.h]: @dev-gh please avoid style check in your PR

File: src/components/can_cooperation/src/can_module.cc [hunk 6]
```diff
@@ -226,76 +226,83 @@ functional_modules::ProcessResult CANModule::HandleMessage(
...
-      break;
-    }
-    case application_manager::MessageType::kRequest: {
-      if (function_name == functional_modules::hmi_api::sdl_activate_app) {
-        msg->set_protocol_version(application_manager::ProtocolVersion::kHMI);
-        return ModuleHelper::ProcessSDLActivateApp(value, *this);
-      }
-      return ProcessResult::CANNOT_PROCESS;
-    }
+    // Disabled
```
Comment (LuxoftAKutsan) [inline src/components/can_cooperation/src/can_module.cc]: @dev-gh please make it as TODO with description why it is disabled

File: src/components/can_cooperation/include/can_cooperation/commands/get_interior_vehicle_data_capabilities_request.h (hunk 1 above)
Comment (dev-gh) [reply src/components/can_cooperation/include/can_cooperation/commands/get_interior_vehicle_data_capabilities_request.h:90]: @AByzhynar 
Currently there is no clarification from customer on format expected for fallback capabilities, so I would leave these parts untill further information will be provided and than change that accordingly.

File: src/components/can_cooperation/src/commands/get_interior_vehicle_data_capabilities_request.cc (hunk 2 above)
Comment (dev-gh) [reply src/components/can_cooperation/src/commands/get_interior_vehicle_data_capabilities_request.cc:161]: @AByzhynar 
Currently there is no clarification from customer on format expected for fallback capabilities, so I would leave these parts untill further information will be provided and than change that accordingly.

File: src/components/can_cooperation/src/validators/get_interior_vehicle_data_capabilities_response_validator.cc (hunk 3 above)
Comment (dev-gh) [reply src/components/can_cooperation/src/validators/get_interior_vehicle_data_capabilities_response_validator.cc]: @AByzhynar 
Done with 40ce3c6

**Review by LitvinenkoIra — APPROVED** [review]

**Review by AByzhynar — APPROVED** [review]

**Review by LuxoftAKutsan — COMMENTED** [review]
@dev-gh changes rebased and cherry picked

**Pull request closed without merging**